project/auth/routes.py

```diff
@@ -0,0 +1,179 @@
+# Modules from Flask Library
+from flask import Flask, render_template, request, flash, Blueprint
+from flask import session as login_session
+from flask import make_response, redirect, jsonify, url_for
+# Access database
+from project.models import Base, Category, Product, User
+from project.db import DBConnector
+# Services
+from project.services.categories import CategoryService
+from project.services.config import ConfigService
+
+import random
+import string
+import httplib2
+import json
+import requests
+
+# Authentification modules
+from oauth2client.client import flow_from_clientsecrets
+from oauth2client.client import FlowExchangeError
+
+
+auth = Blueprint('auth', __name__)
+
+# Connect to the database, create session
+session = DBConnector().get_session()
+
+# Instantiate services
+category_service = CategoryService()
+config_service = ConfigService()
+
+# Get Client ID
+CLIENT_ID = config_service.get_setting('client_id')
+
+
+@auth.route('/login')
+def showLogin():
+    """App route function to display login page."""
+    state = ''.join(random.choice(string.ascii_uppercase + string.digits)
+                    for x in range(32))
+    login_session['state'] = state
+    return render_template('login.html',
+                           STATE=state,
+                           categories=category_service.get_all_categories(),
+                           CLIENT_ID=CLIENT_ID)
+
+
+@auth.route('/googleConnect', methods=['POST'])
+def googleConnect():
+    """Connect via Google Account and fetch User info."""
+    # Validate state token
+    if request.args.get('state') != login_session['state']:
+        response = make_response(json.dumps('Invalid state parameter.'), 401)
+        response.headers['Content-Type'] = 'application/json'
+        return response
```

Comment:
This pattern is repeated over 10 times in this file. When you see something like this, it's always a good candidate to refactor your code and [DRY](http://www.thejavageek.com/2015/04/10/dont-repeat-yourself-principle/) it.
Additionally, it's better (according to code conventions) to [use constants for repeated and/or static values](https://www.programiz.com/python-programming/variables-constants-literals#constants). Response statuses are a good example.
In this particular case, you could do something like this:

```python
HTTP_STATUS_CODE_OK = 200
HTTP_STATUS_CODE_UNAUTHORIZED = 401
HTTP_STATUS_CODE_ERROR = 500

def make_json_response(data, status):
    response = make_response(data, status)
    response.headers['Content-Type'] = 'application/json'
    return response
```

and then
```python
@auth.route('/googleConnect', methods=['POST'])
def googleConnect():
    """Connect via Google Account and fetch User info."""
    # Validate state token
    if request.args.get('state') != login_session['state']:
        return make_json_response('Invalid state parameter.', HTTP_STATUS_CODE_UNAUTHORIZED)
```

_P.S. as far as I remember, you don't need to do `json.dumps` for simple data types like strings or ints._